### PR TITLE
Update descriptor.proto.

### PR DESCRIPTION
**Pull request by jakobr-google**
Updated descriptor.proto and plugin.proto with copies from google/protobuf repo, version 3.3.0.

Moved the files to a separate directory, and regenerated the dart code.

This change unblocks the gRPC stub generation plugin, and is also necessary for the proto3 work.

### CODE REVIEW

File: lib/src/dart_options.pb.dart [hunk 1]
```diff
@@ -1,8 +1,12 @@
 ///
 //  Generated code. Do not modify.
 ///
+// @ignoreProblemForFile non_constant_identifier_names
```
Comment (skybrian) [inline lib/src/dart_options.pb.dart]: Maybe this is a separate patch, but this should probably be switched to the new syntax: `// ignore_for_file`

More: https://github.com/dart-lang/sdk/search?utf8=%E2%9C%93&q=ignore_for_file&type=

**Comment by skybrian** [comment]
Since the generated files are checked in, it's probably okay, but do the instructions in the README about which protoc versions are supported need to be updated? Here's what it currently says: `To compile a .proto file, you must use the 'protoc' command which is installed separately. Protobuf 2.6.1 or above is recommended. Many features may still work as far back as Protobuf 2.5.0 but we are no longer testing this.`

Also, the patch description will be more helpful later if it names the specific version copied, not just "latest".

File: lib/src/dart_options.pb.dart (hunk 1 above)
Comment (jakobr-google) [inline lib/src/dart_options.pb.dart]: I might as well do it now. Coming up.

**Comment by jakobr-google** [comment]
Thanks for the review!

Updated patch description.

We now require at least protoc 3.0.0 to compile descriptor.proto, so I'll update the README to require that. As you say, since we've checked in the generated files, it's probably still possible to use the plugin with older protoc versions, but we might as well update the requirement since there are no guarantees.

**Comment by skybrian** [comment]
LGTM

**Pull request merged**